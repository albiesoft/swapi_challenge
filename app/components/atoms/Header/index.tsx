import { FC } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { Layout, Menu } from 'antd';

const Header: FC = () => {
  const { Header } = Layout;
  const { Item } = Menu;

  const { query } = useRouter();

  const selectedKey = query.name ? query.name : '/';

  return (
    <Header>
      <Menu
        theme="light"
        mode="horizontal"
        defaultSelectedKeys={[selectedKey as string]}
        style={{ lineHeight: '64px' }}
      >
        <Item key="/">
          <Link href="/">
            <a>Home</a>
          </Link>
        </Item>
        <Item key="tatooine">
          <Link href="/planets/[name]" as="/planets/tatooine">
            <a>Tatooine</a>
          </Link>
        </Item>
        <Item key="alderaan">
          <Link href="/planets/[name]" as="/planets/alderaan">
            <a>Alderaan</a>
          </Link>
        </Item>
        <Item key="yavin">
          <Link href="/planets/[name]" as="/planets/yavin">
            <a>Yavin</a>
          </Link>
        </Item>
      </Menu>
      <style jsx>{`
        background: #fff;
      `}</style>
    </Header>
  );
};

export default Header;
