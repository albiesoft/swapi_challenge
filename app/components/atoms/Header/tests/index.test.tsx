import { shallow, ShallowWrapper } from 'enzyme';
import * as nextRouter from 'next/router';
import Header from '..';

describe('<Header />', () => {
  test('It renders correctly', () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({
      route: '/',
      query: {
        name: 'planet'
      }
    }));
    
    const target: ShallowWrapper = shallow(<Header />);
    expect(target).toMatchSnapshot();
  });
});
