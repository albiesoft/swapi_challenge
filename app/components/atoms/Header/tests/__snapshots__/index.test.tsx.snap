// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> It renders correctly 1`] = `
<Header>
  <Menu
    defaultSelectedKeys={
      Array [
        "planet",
      ]
    }
    mode="horizontal"
    style={
      Object {
        "lineHeight": "64px",
      }
    }
    theme="light"
  >
    <MenuItem
      key="/"
    >
      <Link
        href="/"
      >
        <a>
          Home
        </a>
      </Link>
    </MenuItem>
    <MenuItem
      key="tatooine"
    >
      <Link
        as="/planets/tatooine"
        href="/planets/[name]"
      >
        <a>
          Tatooine
        </a>
      </Link>
    </MenuItem>
    <MenuItem
      key="alderaan"
    >
      <Link
        as="/planets/alderaan"
        href="/planets/[name]"
      >
        <a>
          Alderaan
        </a>
      </Link>
    </MenuItem>
    <MenuItem
      key="yavin"
    >
      <Link
        as="/planets/yavin"
        href="/planets/[name]"
      >
        <a>
          Yavin
        </a>
      </Link>
    </MenuItem>
  </Menu>
  <style>
    
        background: #FFF;
      
  </style>
</Header>
`;
